src/shared/components/query/QueryStoreUtils.ts

```diff
@@ -52,13 +52,19 @@ export function queryParams(nonMolecularProfileParams:NonMolecularProfileQueryPa
 
 export function nonMolecularProfileParams(store:QueryStore):NonMolecularProfileQueryParams {
     const selectedStudyIds = store.allSelectedStudyIds;
+    let case_ids: string;
```

Comment:
this seems funky to me.  i would think that if there's a custom caseCase set call that isn't complete, then all of caseIds is pending and user should be waiting for that.  i.e. why show them caseIds preliminarily if custom caseIds is pending and might wipe caseIds out?

Reply:
you're right, I should wait until the custom case set call completes. I'll make it work like that

Reply:
We have an issue to clean some of this stuff up, a lot of things that are computed should prolly just be readonly and await the response: https://github.com/cBioPortal/cbioportal/issues/3771